### PR TITLE
 Simpily demo, add paddle.default_main_program() and paddle.default_startup_program()

### PR DESCRIPTION
**Pull request by helinwang**
- Removed all main_program and startup_program in the demo.
- Using paddle.default_main_program() hides the implementation detail (e.g., using g_main_program) from the user, we can change the implementation in the future much easier.

Fixes: https://github.com/PaddlePaddle/Paddle/issues/5614 https://github.com/PaddlePaddle/Paddle/issues/5305 https://github.com/PaddlePaddle/Paddle/issues/5301

### CODE REVIEW

File: python/paddle/v2/__init__.py [hunk 1]
```diff
@@ -35,8 +35,12 @@
 import image
 import model
 import paddle.trainer.config_parser as cp
+import paddle.v2.framework.framework.default_startup_program as default_startup_program
```
Comment (jacquesqiao) [inline python/paddle/v2/__init__.py]: v2这个应该不用修改，目前重构的代码都在 framework下

Comment (helinwang) [reply python/paddle/v2/__init__.py]: 嗯，是这样用吗：`paddle.framework.default_startup_program()`？

Comment (dzhwinter) [reply python/paddle/v2/__init__.py]: Will we publish our new API under `framework`? It is a little strange that our refactorization API all put into framework directory.

File: python/paddle/v2/framework/framework.py [hunk 2]
```diff
@@ -562,3 +562,9 @@ def __init__(self, block, shape, dtype, **kwargs):
 # program is a global instance.
 g_main_program = Program()
 g_startup_program = Program()
+
+def default_startup_program():
+    return g_startup_program
+
+def default_main_program():
```
Comment (dzhwinter) [inline python/paddle/v2/framework/framework.py]: Why we need to wrap the main program into a global function....

Comment (helinwang) [reply python/paddle/v2/framework/framework.py]: Here explains: https://github.com/PaddlePaddle/Paddle/issues/5614

Comment (dzhwinter) [reply python/paddle/v2/framework/framework.py]: okay..

**Pull request merged**